src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/CodeActionEndpoint.cs

```diff
@@ -249,7 +249,21 @@ internal async Task<IEnumerable<RazorCodeAction>> GetCSharpCodeActionsFromLangua
                 return Array.Empty<RazorCodeAction>();
             }
 
-            var newRequest = context.Request with { Range = projectedRange };
+            var newContext = context.Request.Context;
+            if (context.Request.Context is OmniSharpVSCodeActionContext omniSharpContext &&
```

Comment:
❓Does this need to be a check, or can I just cast to `OmniSharpVSCodeActionContext`?